pandas/plotting/_matplotlib/core.py

```diff
@@ -1390,6 +1407,38 @@ def _get_c_values(self, color, color_by_categorical: bool, c_is_column: bool):
             c_values = c
         return c_values
 
+    def _are_valid_colors(self, c_values: np.ndarray | list):
```

Comment:
In what instances is `c_values` a list? Might be misreading but would be better if we only worked with a pd.Series and could call .unique on that, instead of checking every single value in a loop

Reply:
Updated to take a `pd.Series`, not `np.ndarray | list`